source/Ship.cpp

```diff
@@ -2263,36 +2263,48 @@ int Ship::Scan()
 	if(!cargoDistance && !outfitDistance)
 		return 0;
 	
-	// Scanning speed also uses a square root, so you need four scanners to get
-	// twice the speed out of them.
-	double cargoSpeed = sqrt(attributes.Get("cargo scan speed"));
+	// Scanning speed does not use a square root, so that increasing both power and speed 
+	// proportionally will rise both range and speed. (The longer the range, the slower the scanner.)
+	// Multiply by 100 so that we can simply divide speed by power.
+	double cargoSpeed = 100. * attributes.Get("cargo scan speed") / cargoDistance;
 	if(!cargoSpeed)
-		cargoSpeed = 1.;
-	double outfitSpeed = sqrt(attributes.Get("outfit scan speed"));
+		cargoSpeed = 100. / cargoDistance;
+	
+	double outfitSpeed = 100. * attributes.Get("outfit scan speed") / outfitDistance;
 	if(!outfitSpeed)
-		outfitSpeed = 1.;
+		outfitSpeed = 100. / outfitDistance;
 	
 	// Check how close this ship is to the target it is trying to scan.
 	double distance = (target->position - position).Length();
 	
+	// Check the target's outfit and cargo space, a larger ship takes longer to scan.
+	// Normalized around 200 tons of cargo/outfit space.
+	double outfits = target->baseAttributes.Get("outfit space") / 200.;
+	double cargo = target->attributes.Get("cargo space") / 200.;
```

Comment:
There are four ways I saw I could do this-

**base-attribute outfit space and base-attribute cargo space**. This would return the hull's original values, and make scanning times essentially an inherent trait of hulls. Scan times would not be impacted by outfits installed.

**base-attribute outfit space and attribute cargo space**. This would return the hull's original outfit space, but its current cargo space. Because cargo space isn't actually modified by cargo being held, cargo scan times can be increased by cargo expansions, but not by loading up. Outfit space is still tied to the hull, meaning that packing your ship full of outfits (reducing attributes.Get("outfit space") close to zero) will not reduce outfit scan times. This is the solution I picked, because the two attributes behave as similarly as possible here.

**attribute outfit space and base-attribute cargo space**. This one hits both issues I avoid with the above solution- cargo expansions do not impact cargo scan time, but installing ordinary outfits DOES decrease outfit scan time.

**attribute outfit space and attribute cargo space**. This is the solution I picked first, because I hadn't considered fetching a ship's base attributes. It indeed runs into the issue that a ship with 0 outfit space remaining will be instantly scanned.

As I understand it, your solution would be much more rigorous, by taking base outfit space, plus all the outfit space added by outfits, while ignoring outfits which subtract outfit space? I feel like looping through the outfit list of the target ship could get computationally-expensive, but it _would_ let outfit expansions increase outfit scan times... (I am also not sure I know the best technique to loop through the target's outfit list in the first place)

Sorry this one's a huge rambling mess, I did think this one through, I seem to have forgotten including justification in a comment.